color_balance/colorimage.py

```diff
@@ -89,95 +71,67 @@ def convert_to_colorimage(cimage, band_indices=None,
     out_bands = []
     for band in bands:
-        # If 16-bit, convert to 8-bit, assuming 16-bit image is using the
-        # entire bit depth unless bit depth is given
-        if band.dtype == numpy.uint16:
-            if bit_depth is not None:
-                band = (band / (bit_depth / 2 ** 8))
-            else:
-                band = (band / 2 ** 8)
-        elif band.dtype != numpy.uint8:
-            raise ImagePropertyException(
-                "Band type is {}, should be {} or {}".
-                format(band.dtype,
-                       numpy.uint8, numpy.uint16))
-
-        # Mask out values above or below the 8-bit extents
-        # This is only necessary if bit depth is given
-        clip_max = band > 255
-        cmask[clip_max] = 0
-        clip_min = band < 0
-        cmask[clip_min] = 0
-        out_bands.append(band.astype(numpy.uint8))
-
-    cimg = rgb_bands_to_colorimage(out_bands)
-    return cimg, cmask
-
-
-def rgb_bands_to_colorimage(rgb_bands):
-    return cv2.merge(list(reversed(rgb_bands)))
+        out_bands.append(band)
 
+    # Mask out values above or below the 8-bit extents
+    # This is only necessary if bit depth is given
+    clip_max = band > 255
+    cmask[clip_max] = 0
+    clip_min = band < 0
+    cmask[clip_min] = 0
 
-def colorimage_to_rgb_bands(c_img):
-    return reversed(cv2.split(c_img))
+    cimg = np.dstack(out_bands[::-1])
+    return cimg, cmask
 
 
 def get_histogram(band, mask=None):
-    '''Calculate the histogram of a band. If a mask is provided, masked pixels
-    are not considered in the histogram calculation.
+    """
+    Calculate the histogram of a band. If a mask is provided, masked pixels
+    are not considered in the histogram calculation. The band parameter
+    is assumed to represent a 12bit image when the dtype is np.uint16.
 
-    :param numpy array band: source for calculating histogram
-    :param numpy array mask: array containing zero values at locations where
-        pixels should not be considered, 255 elsewhere
-    :returns: count of pixels at each possible intensity value
-    '''
-    if band.max() > 255 or band.min() < 0:
-        raise OutOfRangeException("Band values outside of [0, 255]")
+    :param band: ndarray of the source for calculating histogram
+    :param mask: boolean ndarray containing True values at locations
+                 where pixels should not be considered.
+    :return: count of pixels at each possible intensity value
+    """
 
-    bit_depth = 256
-    hist = cv2.calcHist([band], [0], mask, [bit_depth], [0, bit_depth])
+    if mask is not None:
+        indices = np.where(mask == False)
+        band = band[indices]
 
-    # cv2.calcHist returns a 2D float array
-    # Convert histogram to 1D array of ints
-    int_hist = hist[:,0].astype(numpy.int)
-    return int_hist
+    bins = 255 if band.dtype == np.uint8 else 4095
+    return np.histogram(band.ravel(), bins=bins, range=[0, bins])[0]
```

Comment:
I'm still a bit confused about flatten() vs. ravel().  Does ravel() make a copy of the data or is it still referencing the original with a different organization?  I ask because I'm concerned that the more copies of data we have in memory the larger the memory footprint of jobs, and I didn't think that numpy.historgram() required flattening.